### PR TITLE
raftstore: avoid call Raft status

### PR DESCRIPTION
**Pull request by july2993**
- avoid heavy status call
- tikv-fail: addr arg is required now,or it will just show a unwrap error msg

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -1712,8 +1711,9 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             //                  |-----------------threshold------------ |
             //              first_index                         replicated_index
             let replicated_idx = peer.raft_group
-                .status()
-                .progress
+                .raft
+                .prs()
+                .voters()
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: we also need to handle learners here.

File: src/bin/tikv-fail.rs [hunk 2]
```diff
@@ -49,6 +49,7 @@ fn main() {
         .about("A tool for injecting failures to TiKV and recovery")
         .arg(
             Arg::with_name("addr")
+                .required(true)
```
Comment (siddontang) [inline src/bin/tikv-fail.rs]: please use another PR for this change :-)

**Comment by BusyJay** [comment]
I don't think it's a good idea to access raft directly. Especially when we are about to make raft module as a standalone crate.

**Comment by siddontang** [comment]
@BusyJay 

Maybe we can provide a match_ids function in the raft lib.

File: src/bin/tikv-fail.rs (hunk 2 above)
Comment (july2993) [reply src/bin/tikv-fail.rs]: https://github.com/pingcap/tikv/pull/2602
i have make another PR here

File: src/raftstore/store/store.rs (hunk 1 above)
Comment (july2993) [reply src/raftstore/store/store.rs]: sorry, i am not sure whether we need to handle learners here yet, i read the code before the learner feature been merged these days, but i make sure the the code(master branch now) did'n handle learners too before fire this PR.
agree with @BusyJay,  sounds more work must be do.

but anyway, if we need to handle learners here, somethink must be change here.

Comment (siddontang) [reply src/raftstore/store/store.rs]: we must handle learners later, the learner can also compact the raft log. So you can help us support this right now.

**Comment by ngaut** [comment]
Currently, i am ok with that, we can merge it now, and upgrade when raft crate is ready.

File: src/raftstore/store/store.rs (hunk 1 above)
Comment (july2993) [reply src/raftstore/store/store.rs]: i think if not include learner, compact_idx bigger than the matched index of some learner,
it maybe easy to  trigger the case where leader have to send snapshot to the learner

**Comment by siddontang** [comment]
Cool, LGTM

Now we also use raft directly in raftstore in some places, so we can change all later.

PTAL @BusyJay

**Comment by BusyJay** [comment]
Thanks for your contributions! :beer:

**Pull request merged**